### ISSUE TITLE
Instance variable initialized in method called from `__init__`

### ISSUE THREAD

**Issue by ryan-gunderson** [microsoft/pyright#3447]
The recent modifications to the behavior of _reportUninitializedInstanceVariable_ option (discussed in #3415) causes the following code to give a reportUninitializedInstanceVariable error for `y`:
```
class Foo:
    x: int
    y: int

    def __init__(self):
        self.x = 5
        self.initialize_y()

    def initialize_y(self):
        self.y = 10
```

Here `y` is initialized but in a method called from within `__init__`.  This seems like a very common way to initialize instance variables, especially in larger classes.

**Comment by erictraut** [comment]
@ryan-gunderson, if you want to use reportUninitializedInstanceVariable, you'll need to initialize it directly within __init__. This diagnostic check is among the strictest (it's not even on in "strict" mode) and is necessarily very opinionated. I was reluctant to add it in the first place because it is so limiting and has the potential for so many false positives.

**Comment by JanEricNitschke** [comment]
Just as a data point, i would also love if this special case were added to the check. For me it is currently the only thing blocking me from using it.

I do understand that the check is opinionated, but i saw in another related issue that you might add special cases if there is enough demand, so i would just like to show my interest for this one.
